std/algorithm/mutation.d

```diff
@@ -1420,9 +1420,17 @@ void moveEmplace(T)(ref T source, ref T target) @system
     }
     else
     {
-        // Primitive data (including pointers and arrays) or class -
-        // assignment works great
-        target = source;
+        static if (isStaticArray!T)
+        {
+            for (size_t i = 0; i < source.length; ++i)
+                move(source[i], target[i]);
+        }
+        else
+        {
+            // Primitive data (including pointers and arrays) or class -
+            // assignment works great
+            target = source;
+        }
```

Comment:
I would write the whole thing as:
```D
static if (is(T == struct))
{
    // ...
}
else static if (isStaticArray!T)
    moveEmplaceAll(source[], target[]);
else
    target = source;

```

Reply:
Can't use `moveEmplaceAll` as `source[]` fails the `isInputRange` check

Reply:
Yes, unfortunately the `isInputRange` constraint is too strict for `move[Emplace]{All|Some}` functions, as a range, whose elements are not copy-able is the primary use-case for `move`-ing.

If we adjust `isInputRange` like so 
```diff
@@ -168,7 +168,10 @@
 enum bool isInputRange(R) =
     is(typeof(R.init) == R)
     && is(ReturnType!((R r) => r.empty) == bool)
-    && is(typeof((return ref R r) => r.front))
+    && (
+            is(typeof((return ref R r) => r.front)) ||
+            is(typeof((return ref R r) => &r.front()))
+       )
     && !is(ReturnType!((R r) => r.front) == void)
     && is(typeof((R r) => r.popFront));
```

all tests pass.